exposed-dao/src/main/kotlin/org/jetbrains/exposed/dao/EntityCache.kt

```diff
@@ -137,8 +139,23 @@ class EntityCache(private val transaction: Transaction) {
         }
     }
 
+    fun referrers(entityId: EntityID<*>): Map<Column<*>, SizedIterable<*>> {
+        return referrerEntities[entityId]?.mapNotNull { column ->
+            referrers[column]?.get(entityId)?.let { column to it }
+        }?.toMap() ?: emptyMap()
+    }
+
+    fun clear(flush: Boolean = true) {
+        if (flush) flush()
+        data.clear()
+        inserts.clear()
+        updates.clear()
+        clearReferrersCache()
+    }
```

Comment:
I added these helpers as replacements for consumers who were fiddling with the internal data structures directly, including tests.